src/test/shared/telemetry/defaultTelemetryService.test.ts

```diff
@@ -65,7 +68,37 @@ describe('DefaultTelemetryService', () => {
         assert.strictEqual(mockPublisher.flushCount, 0)
         assert.strictEqual(mockPublisher.enqueueCount, 0)
         assert.strictEqual(mockPublisher.enqueuedItems, 0)
-        // and there are no records in memeory
+        // and events are kept in memory
+        assert.strictEqual(service.records.length, 3)
+        // events are, in order, the dummy test event, the start event, and the shutdown event
+        // test event is first since we record it before starting the service
+        assert.strictEqual(service.records[0].namespace, 'name')
+        assert.strictEqual(service.records[1].namespace, 'ToolkitStart')
+        assert.strictEqual(service.records[2].namespace, 'ToolkitEnd')
+    })
+
+    it('events are never recorded if telemetry has been disabled', async () => {
```

Comment:
Can you please add a test for the situation where events are added, and then the state moves from "option not Made" to "enabled"? We want to verify that the queued elements do not zero out.

Reply:
sure